developer_manual/app_publishing_maintenance/release_process.rst

```diff
@@ -0,0 +1,126 @@
+
+At this point you can probably assign a release date (if not set already) on the milestone and close it.
+
+.. _app-versioning:
+
+Versioning
+~~~~~~~~~~
+
+Each update of an app needs a new, higher version number. Before preparing a specific release, the maintainer has to decide what version number the new release will get.
+
+It is highly recommended that apps follow `semantic versioning <https://semver.org/>`_ (semver). This schema allows admins and users to better understand what type of change they can expect when an app update is available.
+
+Applying semver on Nextcloud apps gives three types of updates: major, minor and patch.
+
+Patch updates
+*************
+
+Increment the patch version number when
+
+* A bug gets fixed
+* Translations are added or fixed, but **not** removed
+
+.. tip:: Example: the app is at version 3.7.2. The next patch version will be 3.7.3.
+
+Minor updates
+*************
+
+Increment the minor version number when
+
+* There is a new feature
+* A new major version of Nextcloud will be supported
+* A new major or minor version of PHP will be supported
+* An additional database type is supported
+* Any other change that keeps the app compatible with previously compatible environments (forward compatibility)
+
+.. tip:: Example: the app is at version 3.7.2. The next minor version will be 3.8.0.
+
+Major update
+************
+
+Increment the major version number when
+
+* The app drops support for a major version of Nextcloud, e.g. when Nextcloud 20 support is removed and the app now requires Nextcloud 21 or newer
+* The app drops support for a major or minor version of PHP, e.g. when PHP7.2 is removed and the app now requires PHP7.3 or newer
+* A database type is no longer supported, e.g. when the maintainer decides to stop SQLite support
+* Any other change that makes the app incompatible with a previously compatible environment (breaking change)
+
+.. tip:: Example: the app is at version 3.7.2. The next major version will be 4.0.0.
+
+Change log
+~~~~~~~~~~
+
+If the app `keeps a change log <https://keepachangelog.com/en/1.0.0/>`_ (e.g. CHANGELOG.md in the project root) it's time to update it with all **added**, **changed** and **fixed** items.
+
+.. tip:: Use ``git log v3.7.1..HEAD --pretty=oneline | grep -v tx-robot | grep -v "Merge " | grep -v "Bump "`` on the target release branch to get a compact list of all changes since the v3.7.1 release. Adjust this to your previous release version number.
+
+Pre-releases
+~~~~~~~~~~~~
+
+Optionally maintainers may chose to pre-release their apps on the app store for early adopters and testers. The precise periodization is up to the maintainer. Common periods include **alpha**, **beta** and **rc** (release candidate) epochs.
+
+The release process is identical to the one of a final release, only the version number has a suffix.
```

Comment:
Maybe a note on **not** using the nightly flag on the app store for pre-releases would be good, as that might be a pitfall one (like me) ran into. As Joas told me, having the semver pre-release tag is enough.

Reply:
Good point. That should (also) go on https://docs.nextcloud.com/server/stable/developer_manual/app_publishing_maintenance/publishing.html I think.